src/wallet/wallet.cpp

```diff
@@ -4315,6 +4316,76 @@ bool CWallet::BackupWallet(const std::string& strDest) const
     return GetDatabase().Backup(strDest);
 }
 
+bool CWallet::QueuedTransactionExists(const uint256& txid)
+{
+    LOCK(cs_wallet);
+    return queuedTransactionMap.count(txid) > 0;
+}
+
+bool CWallet::WriteQueuedTransaction(const uint256 &txid, const CMutableTransaction &tx)
+{
+    LOCK(cs_wallet);
+    CQueuedTransaction data;
+    data.setTx(tx);
+
+    const bool success = WalletBatch(GetDatabase()).WriteQueuedTransaction(txid, data);
+    if(success)
+        queuedTransactionMap[txid] = data;
+
+    return success;
+}
+
+bool CWallet::EraseQueuedTransaction(const uint256& txid)
+{
+    LOCK(cs_wallet);
+    const bool success = WalletBatch(GetDatabase()).EraseQueuedTransaction(txid);
+    if(success)
+        queuedTransactionMap.erase(txid);
+    return success;
+}
+
+bool CWallet::GetQueuedTransaction(const uint256 &txid, CQueuedTransaction *data)
+{
+    LOCK(cs_wallet);
+    auto it = queuedTransactionMap.find(txid);
+    if (it == queuedTransactionMap.end())
+        return false;
+    if (data != nullptr)
+        (*data) = it->second;
+    return true;
+}
+
+void CWallet::EffectTransactionQueue()
+{
+    // This is called when we get a new block to deal with queued transactions.
+    if (chain().isInitialBlockDownload())
+        return;
+    // We only do this for fresh blocks. Otherwise, we might trigger way too early.
+    std::set<uint256> to_dequeue;
+    // Kludge; we can't remove items in a map while iterating over it.
+    LOCK(cs_wallet);
+    for (const auto& i : queuedTransactionMap)
+    // For each transaction in the queue...
+    {
+        const uint256 txid = i.first;
+        const CQueuedTransaction qtx = i.second;
+        const CMutableTransaction tx = qtx.getTx();
```

Comment:
Previously, you used all references with code like this.  I think references are better, but in any case it should be consistent.

Reply:
Done.